janus_server/src/bin/janus_cli.rs

```diff
@@ -189,6 +242,49 @@ async fn create_datastore_key(
     Ok(())
 }
 
+/// Decode the contents of `message_file` as a DAP message with `media_type`, returning the decoded
+/// object.
+fn decode_dap_message(message_file: &str, media_type: &str) -> Result<Box<dyn Debug>> {
+    let mut reader = if message_file.eq("-") {
+        Box::new(stdin()) as Box<dyn Read>
+    } else {
+        Box::new(File::open(message_file)?) as Box<dyn Read>
+    };
+
+    let mut message_buf = vec![];
+    reader.read_to_end(&mut message_buf)?;
+
+    let mut binary_message = Cursor::new(message_buf.as_slice());
+
+    let decoded = match media_type {
```

Comment:
I opted to only support the top-level messages that go on the wire as opposed to every single struct with a `prio::codec::Decode` implementation, mainly so this match wouldn't get too sprawling and huge.